server/container_create.go

```diff
@@ -412,8 +412,6 @@ func hostNetwork(containerConfig *types.ContainerConfig) bool {
 func (s *Server) CreateContainer(ctx context.Context, req *types.CreateContainerRequest) (res *types.CreateContainerResponse, retErr error) {
 	log.Infof(ctx, "Creating container: %s", translateLabelsToDescription(req.Config.Labels))
 
-	s.updateLock.RLock()
-	defer s.updateLock.RUnlock()
```

Comment:
Can we be sure that everything on top of this stack is properly locked?

Reply:
TBH, I don't know. I was proposing this commit simply because of this observation (unexported RWMutex only ever being read-locked):
```
klitkey1-mobl cri-o $ git grep updateLock HEAD^
HEAD^:server/container_create.go:       s.updateLock.RLock()
HEAD^:server/container_create.go:       defer s.updateLock.RUnlock()
HEAD^:server/sandbox_run_linux.go:      s.updateLock.RLock()
HEAD^:server/sandbox_run_linux.go:      defer s.updateLock.RUnlock()
HEAD^:server/server.go: updateLock sync.RWMutex
```
Unless it is already broken, removing that lock should not break anything.

Reply:
Yeah I agree. There is always the risk that we reveal some missing lower level locks which would cause a regression. I think we should be fine removing those locks. @haircommander WDYT?

Reply:
yeah I agree if we're only ever read locking it it doesn't seem like we need it.